app/jsonimplicits/implicits.scala

```diff
@@ -7,93 +7,87 @@ import play.api.libs.json.JsString
 import agent._
 import play.api.mvc.RequestHeader
 import _root_.model.{Owner, SSA}
+import org.joda.time.DateTime
 
 trait RequestWrites[T] {
   def writes(request: RequestHeader): Writes[T]
 }
 object RequestWrites {
-  def fromWrites[T](implicit delegate:Writes[T]) = new RequestWrites[T] {
-    def writes(request: RequestHeader): Writes[T] = delegate
-  }
+  def fromWrites[T](implicit delegate:Writes[T]): RequestWrites[T] = (_: RequestHeader) => delegate
 }
 
 object joda {
   implicit object dateTimeWrites extends Writes[org.joda.time.DateTime] {
     def writes(d: org.joda.time.DateTime): JsValue = JsString(ISODateTimeFormat.dateTime.print(d))
   }
-  implicit val dateTimeReads = Reads.jodaDateReads("yyyy-MM-dd'T'HH:mm:ss.SSSZZ")
+  implicit val dateTimeReads: Reads[DateTime] = JodaReads.jodaDateReads("yyyy-MM-dd'T'HH:mm:ss.SSSZZ")
 }
 
 object model {
   import joda.dateTimeWrites
 
-  implicit val originWriter = new Writes[Origin] {
-    def writes(o: Origin): JsValue = o.toJson
-  }
+  implicit val originWriter: Writes[Origin] = (o: Origin) => o.toJson
 
-  implicit val securityGroupRefWriter = Json.writes[SecurityGroupRef]
-  implicit val securityGroupRuleWriter = Json.writes[Rule]
-  implicit val securityGroupWriter = Json.writes[SecurityGroup]
+  implicit val securityGroupRefWriter: OWrites[SecurityGroupRef] = Json.writes[SecurityGroupRef]
```

Comment:
Hmmm. For neatness these should probably be `Writes[A]`. IDEA hasn't quite found the intended type (although it's a subtype so will obviously work).

Reply:
✅